lib/Transforms/Scalar/DxilRemoveDeadBlocks.cpp

```diff
@@ -346,8 +346,7 @@ static void EnsureDxilModule(Module *M) {
     return;
   for (Function &F : *M) {
     if (OP::IsDxilOpFunc(&F)) {
-      bool bSkipInit = true; // Metadata is not necessarily valid yet.
-      M->GetOrCreateDxilModule(bSkipInit);
+      M->GetOrCreateDxilModule();
```

Comment:
Is metadata ready at DxilRemoveDeadBlocks?

Reply:
Either we have a DxilModule that doesn't need to be initialized (since we are compiling normally from the start), or we need to construct the DxilModule and load state from metadata because we deserialized something into an llvm module.

In the former case, we don't have metadata, but we do have the necessary state in DxilModule that would be used to emit the metadata.  The DxilModule state is what's needed, not the metadata itself.  Since we already have a DxilModule, that just gets returned, so the bSkipInit param is not used.

In the latter case, we should have metadata capturing the important properties of the module, otherwise we forgot to capture the key state from DxilModule by emitting the metadata before serialization.  If we don't have the metadata, we lost information that answers important questions, such as whether this module is min-precision mode or exact-low-precision mode.

Originally, this was written with `bSkipInit=true` because of the `delete_constant_dce.ll` IR test which was generated by dumping a module without emitting metadata first.  But that's not the right way to produce IR for a pass test, since it loses important information contained in DxilModule needed to interpret the IR.  This is why I wrote `ExtractIRForPassTest.py` to make it easy to extract IR with metadata in the state before a particular pass to turn into a test (then I checked that in separately).

**TL;DR: We do not need bSkipInit, unless we are already in an invalid state.**